Modules/Scripted/Home/HomeLib/segmentation_model.py

```diff
@@ -2,44 +2,70 @@
     """
-
-    model_dict = torch.load(load_path, map_location=torch.device('cpu'))
-
-    self.seg_net = model_dict['model']
-    self.learning_rate = model_dict['learning_rate']
-    self.training_losses = model_dict['training_losses']
-    self.validation_losses = model_dict['validation_losses']
-    self.epoch_number = model_dict['epoch_number']
-    self.best_validation_loss = model_dict['best_validation_loss']
-    self.best_validation_epoch = model_dict['best_validation_epoch']
-    self.image_size = model_dict['image_size']
-
-    # Transforms a given image to the input format expected by the segmentation network
-    self.transform = monai.transforms.Compose([
-      monai.transforms.CastToType(dtype=np.float32), # TODO dtype should have been included in the model_dict
-      monai.transforms.AddChannel(),
-      monai.transforms.Resize(
-        spatial_size=(self.image_size,self.image_size),
-        mode = 'bilinear',
-        align_corners=False
-      ),
-      monai.transforms.ToTensor()
-    ])
-
-    self.seg_post_process = SegmentationPostProcessing()
+    self.model_source = self.ModelSource.USE_PTH
+
+    self.load_pth_path = load_pth_path
+    # For save_zip_path, remove trailing .pth if present; append .zip
+    self.save_zip_path = re.sub(r"\.pth$", "", self.load_pth_path) + ".zip"
+
+    if self.model_source == self.ModelSource.USE_PTH:
+      model_dict = torch.load(self.load_pth_path, map_location=torch.device('cpu'))
+
+      self.seg_net = model_dict['model']
+      self.learning_rate = model_dict['learning_rate']
+      self.training_losses = model_dict['training_losses']
+      self.validation_losses = model_dict['validation_losses']
+      self.epoch_number = model_dict['epoch_number']
+      self.best_validation_loss = model_dict['best_validation_loss']
+      self.best_validation_epoch = model_dict['best_validation_epoch']
+      self.image_size = model_dict['image_size']
+
+      # Transforms a given image to the input format expected by the segmentation network
+      self.transform = monai.transforms.Compose([
+        monai.transforms.CastToType(dtype=np.float32), # TODO dtype should have been included in the model_dict
+        monai.transforms.AddChannel(),
+        monai.transforms.Resize(
+          spatial_size=(self.image_size,self.image_size),
+          mode = 'bilinear',
+          align_corners=False
+        ),
+        monai.transforms.ToTensor()
+      ])
+
+      self.seg_post_process = SegmentationPostProcessing()
+
+    if self.model_source in (self.ModelSource.USE_LOCAL_ZIP, self.ModelSource.USE_DOCKER_ZIP) and not os.path.exists(self.save_zip_path):
+      # Write out a TorchScript version of the model, for use in MONAI Deploy.
+      model_dict = torch.load(self.load_pth_path, map_location=torch.device('cpu'))
+      seg_net = model_dict['model']
+      # set dropout and batch normalization layers to evaluation mode before running
+      # inference
+      tmp = seg_net.eval()
```

Comment:
Hmm are you sure it was outputting statements to the console? I don't think this should print anything. Maybe it only happened because you were in an interactive python session? (you know that feature that interactive sessions often have where they print out the return value of your last expression if you don't assign it to anything)

Reply:
Yes, it was an interactive Python session.  I will remove the assignment.